### ISSUE TITLE
wss:// gets redirected to IPNS

### ISSUE THREAD

**Issue by mkg20001** [ipfs-shipyard/ipfs-companion#469]
When accessing https://libp2p-nodetrust.tk/demo it gets properly redirected to  http://127.0.0.1:8080/ipns/libp2p-nodetrust.tk/demo/
When launching the demo it connects to wss://libp2p-nodetrust.tk which should not be redirected, yet it gets redirected to http://127.0.0.1:8080/ipns/libp2p-nodetrust.tk/demo/ which breaks the demo

Fix: Detect and don't modify websocket requests

**Comment by mkg20001** [comment]
Just tested it, the results look weird. i'll retest it on a clean firefox profile again.
```
| -                 | ff  | chrome |
|-------------------|-----|--------|
| with dnslink      | [ ] | [x]    |
| without dnslink   | [ ] | [x]    |
| without companion |     | [x]    |
```

**Comment by mkg20001** [comment]
On the nightly indeed everything seems to work...
```
| -                 | ff  | chrome |
|-------------------|-----|--------|
| with dnslink      | [x] | [x]    |
| without dnslink   | [x] | [x]    |
| without companion | [x] | [x]    |
```

**Comment by lidel** [comment]
I was able to replicate it yesterday with Firefox 59, but _only when dnslink was enabled_.
@mkg20001 did you change the web app by any chance? I am no longer able to replicate the bug today – page seems to work file with and without dnslink. 

Anyway,  some plausible explanations: 

- The extension was breaking the WSS  in by replacing initial HTTP GET that was supposed to end with `101 Switching Protocols` (and upgrade to websocket connection) with a plain HTTP 307 redirect to IPNS page at localhost  (I am low key assuming your app used implicit hostname for websocket handshake here)
- The HTML and JS loaded from IPNS was an older (broken?) version than the one at https://libp2p-nodetrust.tk/demo. 
- Upstream difference between API implementation across browser vendors.  Namely the handshake may be included 

I agree with general suggestion. To be on the safe side  _the extension should explicitly skip requests related to WSS_ (HTTP2IPFS gateways do not support WSS anyway). That way we will be immune to ephemeral bugs caused by browsers including/excluding websocket handshake  from `<all_urls>`. 

FYSA right now, both Chrome and Firefox (nightly and stable) include it in mentioned scope and pass it to both hooks: 

| Firefox | Chrome |
| --- | --- |
| ![screenshot_1](https://user-images.githubusercontent.com/157609/39519547-776be994-4e07-11e8-8103-54c865caced2.png)  |  ![screenshot_6](https://user-images.githubusercontent.com/157609/39519490-3b92660a-4e07-11e8-8997-e478f141f5df.png) | 

I will prepare PR shortly.

**Comment by lidel** [comment]
@mkg20001 should be fixed in latest beta ([v2.2.2.9120](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.2.2rc1))
